internal/batch/batch_manager.go

```diff
@@ -175,10 +177,14 @@ func (bm *batchManager) getProcessor(txType fftypes.TransactionType, msgType fft
 	return processor, nil
 }
 
-func (bm *batchManager) assembleMessageData(msg *fftypes.Message) (data []*fftypes.Data, err error) {
+func (bm *batchManager) assembleMessageData(processor *batchProcessor, msg *fftypes.Message) (retData fftypes.DataArray, err error) {
+	var cro []data.CacheReadOption
+	if processor.conf.DispatcherOptions.BatchType == fftypes.BatchTypeBroadcast {
```

Comment:
Could consider just passing the BatchType into this method, instead of the processor itself...